service/history/api/signalworkflow/api.go

```diff
@@ -106,12 +106,15 @@ func Invoke(
 			if request.GetRequestId() != "" {
 				mutableState.AddSignalRequested(request.GetRequestId())
 			}
-			if _, err := mutableState.AddWorkflowExecutionSignaled(
+			_, err := mutableState.AddWorkflowExecutionSignaledEvent(
 				request.GetSignalName(),
 				request.GetInput(),
 				request.GetIdentity(),
 				request.GetHeader(),
-				request.GetSkipGenerateWorkflowTask()); err != nil {
+				request.GetSkipGenerateWorkflowTask(),
+				parentExecution,
```

Comment:
I would rename this to `externalWorkflowExecution` for consistency. Especially given a fact that it is comming from `req.ExternalWorkflowExecution`.